src/server/recaptcha/client.ts

```diff
@@ -86,3 +87,33 @@ export async function createRecaptchaAssesment({
     throw throwBadRequestError('Provided token does not match performed action');
   }
 }
+
+type SiteVerifyResponse = z.infer<typeof siteVerifyResponseSchema>;
+const siteVerifyResponseSchema = z.object({
+  success: z.boolean(),
+  challenge_ts: z.coerce.date().optional(),
+  hostname: z.string().optional(),
+  'error-codes': z.array(z.string()),
+  action: z.string().optional(),
+  cdata: z.string().optional(),
+});
+
+export async function verifyCaptchaToken({ token, ip }: { token: string; ip?: string }) {
+  const result = await fetch('https://challenges.cloudflare.com/turnstile/v0/siteverify', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      secret: env.CLOUDFLARE_TURNSTILE_SECRET,
+      response: token,
+      remoteip: ip,
+    }),
+  });
+  if (!result.ok) throw throwBadRequestError('No response from captcha service');
+
+  const outcome = (await result.json()) as SiteVerifyResponse;
+  if (outcome.success) {
+    return true;
+  } else {
+    throw throwBadRequestError('Unable to verify captcha token');
+  }
+}
```

Comment:
Since this is not recaptcha, I'd move this to a new file 👀 `turnstile/client.ts` or something of the sort. recaptcha is a google product.